test/node-api/test_async_cleanup_hook/binding.cc

```diff
@@ -0,0 +1,71 @@
+#define NAPI_EXPERIMENTAL
```

Comment:
This should be a C file, not a C++ file.

Reply:
Is there a particular reason for that? This was based on another C++ test file, and the code would be a lot more convoluted in C.

Reply:
The reason is that the code might be used with other N-API implementations which may or may not require C++. For example, https://github.com/jerryscript-project/iotjs/ is pure C and implements some Node.js APIs like async work. It would be encouraging if the tests built against such outside implementations without having to add a C++ compiler to their build system.

AFAICT it's only a couple of `static_cast`s and two lambdas which would have to be moved out of the body of their containing functions.

Reply:
@gabrielschulhof I’ve converted this to a C file.

However, I would strongly encourage you to address this yourself for other tests instead of expecting people to do the work as part of the PR review process. It’s generally not a great situation for contributors if you can’t base a new test off existing tests.